waspc/data/Cli/templates/new/.wasproot

```diff
@@ -0,0 +1 @@
+File marking the root of Wasp project.
```

Comment:
Should we just get rid of this file and start looking for a .wasp directory? It means we'd have to create an empty .wasp directory whenever the user initializes a project, but I think it's worth it.

Reply:
Not sure if we want to remove it (since we cannot remember the good reason it was there, maybe that is ok :D), but as of now a new project won't compile without it (was just trying it out). Additionally, compile works but trying `wasp-cli start` errors out without a server dir (`wasp-cli: /tmp/splitTest/server/: getDirectoryContents:openDirStream: does not exist (No such file or directory)`)

Reply:
I think we remembered the reasons why we're using .wasproot to detect the root of the project:
- We weren't using main.wasp because, one day, we could support multiple .wasp files nested in different directories.
- We weren't using the hidden `.wasp` directory because `wasp new` doesn't create it and also because we didn't want to break the CLI if the users deleted their `.wasp` directory.

The second reason for not using the .wasp directory sure is a nice property, but maybe it's worth compromising for the purposes of removing an extra file (after all, they could delete that one too).

Reply:
Ah, thanks for the reminders! Makes sense.

We would also need to modify `wasp clean` to just empty, but not entirely remove the .wasp dir, like it does now. I'd probably vote to leave it as is for a few reasons:
- no extra work to do so :)
- it feels more intentional to have a flag file to indicate where the project root is, as opposed to a build output dir (which we may change in the future)
- in the future, I can also imagine we will want to add additional Wasp-related metadata to `.wasproot` as well (possibly renaming it, but serving dual purposes), for example, things like deployment info from prior `wasp deploy` runs or something

Just my 2c, don't feel too strongly about either direction, but wanted to share the counterpoints anyways.

Reply:
Yeah, makes sense. It can stay. I thought "Git does it, so could we", but, to be fair, Git is much more robust than Wasp at this stage (i.e., users will rarely feel like they need to "reset" things by deleting the entire directory).